.github/workflows/build.yaml

```diff
@@ -46,3 +27,18 @@ jobs:
       shell: nix develop .#ci -c bash -e {0}
       run: |
           make
+
+  emulate_test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup nix
+      uses: ./.github/actions/setup-nix
+    - name: Emulate on QEMU
+      shell: nix develop .#ci -c bash -e {0}
+      run: |
+          make emulate
+
+          for file in elf/*.elf; do
+            make emulate ELF_FILE="$file"
```

Comment:
This is not going to fail if the tests output `ERROS keys`, right? I don't see how this could ever fail. 
Can you please make sure this actually catches test failures?

Reply:
Oh, yes, I'm aware of it, I was thinking about first get this PR merged, then fix this in another PR as I haven't looked into the test output format yet.

But yeah it's better to do it in this PR if I only need to check for `ERROR KEYS` for now

Reply:
It is probably more robust to check for OK occurring 30 times.

Reply:
good idea